libbeat/publisher/pipeline/processor.go

```diff
@@ -91,10 +91,13 @@ func newProcessorPipeline(
 	// setup 5: client processor list
 	processors.add(localProcessors)
 
-	// setup 6: add beats metadata
+	// setup 6: add beats and host metadata
 	if meta := global.beatsMeta; len(meta) > 0 {
 		processors.add(makeAddFieldsProcessor("beatsMeta", meta, needsCopy))
 	}
+	if meta := global.hostMeta; len(meta) > 0 {
+		processors.add(makeAddFieldsProcessor("hostMeta", meta, needsCopy))
```

Comment:
This is safe, as needsCopy is true if any global processors are configured.

We can also 'reduce' some copies by introducing `needsGlobalCopy := global.alwaysCopy || global.processors != nil` and `needsCopy := needsGlobalCopy || local.processors != nil`.